hydroflow_datalog/src/join_plan.rs

```diff
@@ -127,19 +138,60 @@ pub fn expand_join_plan(
                 .or_insert_with(Counter::new)
```

Comment:
I realized you could just use `RangeFrom<usize>` and `0..` instead of the custom counter